sqlglot/parser.py

```diff
@@ -1601,11 +1601,40 @@ def _parse_insert(self) -> exp.Expression:
             exists=self._parse_exists(),
             partition=self._parse_partition(),
             expression=self._parse_ddl_select(),
+            on_conflict=self._parse_on_conflict(),
             returning=self._parse_returning(),
             overwrite=overwrite,
             alternative=alternative,
         )
 
+    def _parse_on_conflict(self) -> t.Optional[exp.Expression]:
+        if not self._match(TokenType.ON):
```

Comment:
this should be

conflict = self._match_seq("ON", "CONFLICT")
duplicate = self._match_seq("ON", "DUPLICATE KEY")

if not conflict or duplicate:
   return None